rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -252,24 +288,26 @@ fn keyring_entry() -> Result<keyring::Entry> {
 
 #[derive(Clone)]
 struct CallbackHandler {
-    ctlr_tx: CtlrTx,
     logger: Option<file_logger::Handle>,
+    notify_controller: Arc<Notify>,
+    resources: Arc<ArcSwap<Vec<ResourceDescription>>>,
```

Comment:
Why use an `ArcSwap` instead of a `Mutex`?

Reply:
ArcSwap says it `store` is lock-free by default - https://docs.rs/arc-swap/latest/arc_swap/strategy/type.DefaultStrategy.html
So I think that means connlib _should_ never block here, even if the GUI is stuck inside `load` and frozen, connlib will just drop off the new resource list and keep running. When the GUI unfreezes, the next `notified` event will cause it to eventually pick up the new list.

Reply:
You're cloning the instant that you load the resource, so GUI should uninvolved with the `lock`ing mechanism

Reply:
But if I do a Mutex or RwLock, won't the GUI need to lock it to clone? And then if the GUI hung inside that lock (an extreme worst case) then connlib would block waiting to lock it for writes

Reply:
oops, thought I answered to this one, sorry for the delay.

if you `clone` immediately after the `lock` there's no reason for the thread to hang you're never yielding

Reply:
For tasks yes, but in theory the OS thread could be pre-empted, right?